src/sensai/vector_model.py

```diff
@@ -161,39 +163,61 @@ def isFitted(self) -> bool:
         pass
 
 
-class RuleBasedModel(FittableModel, ABC):
-    """
-    Base class for models where the essential prediction logic is based on rules coded by humans
-    and thus does not require fitting. However, the input generation process may use mechanisms
-    such as feature generation or data transformation, which may require fitting.
-    """
-    def __init__(self):
-        super().__init__()
+class ClassificationMixin(ABC):
+    def isRegressionModel(self):
+        return False
 
-    def fit(self, X: pd.DataFrame, Y: pd.DataFrame = None):
-        if self._featureGenerator is not None:
-            self._featureGenerator.fit(X, Y=Y, ctx=self)
-        self._inputTransformerChain.fit(X)
+    @abstractmethod
+    def getClassLabels(self) -> list:
+        pass
 
-    def isFitted(self):
-        return self._prePostProcessorsAreFitted()
+    @abstractmethod
+    def convertClassProbabilitiesToPredictions(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        The implementation will typically make use of the private method of the same name as well as
+        previously saved information about the predictions' column name
+        """
+        pass
 
-    def predict(self, x: pd.DataFrame) -> pd.DataFrame:
+    def _checkPrediction(self, predictionDf: pd.DataFrame, maxRowsToCheck=5):
         """
-        Performs a prediction for the given input data frame
+        Checks whether the column names are correctly set and whether the entries correspond to probabilities
+        """
+        labels = self.getClassLabels()
+        if list(predictionDf.columns) != labels:
+            raise Exception(f"{self} _predictClassProbabilities returned DataFrame with incorrect columns: "
+                            f"expected {labels}, got {predictionDf.columns}")
 
-        :param x: the input data
-        :return: a DataFrame with the same index as the input
+        dfToCheck = predictionDf.iloc[:maxRowsToCheck]
+        for i, (_, valueSeries) in enumerate(dfToCheck.iterrows(), start=1):
+
+            if not all(0 <= valueSeries) or not all(valueSeries <= 1):
```

Comment:
This is an enhancement of the previous check

Reply:
Could come in handy, for sure.